seminars.rkt

```diff
@@ -4,36 +4,39 @@
    gregor
    "templates.rkt")
 
+@;; The seminars are sorted every time this page is built, to separate
+@;; future seminars from past seminars and order them differently.
+
+@(define-struct seminar (anchor title speaker link aff date room abstract bio))
+
+@(define (in-future? sem)
+   (datetime<? (now #:tz "America/New_York") (seminar-date sem)))
+
 @(define id 0)
-@(define (seminar anchor title speaker link aff date room abstract bio)
+@(define (render-seminar sem)
    (set! id (add1 id))
+   (define finished (if (in-future? sem) "" "finished"))
    @list[
-     @div[id: @format["seminar-~a" id] class: "col-md-12 pn-seminar"]{
-       @script/inline[type: "text/javascript"]{
-         if (new Date() >= new Date("@(datetime->iso8601 date)")) {
-            document.getElementById("seminar-@|id|").classList.add("finished");
-         }
-       }
-       @div[id: anchor class: "pn-main-informations"]{
-         @a[onclick: "return false;" class: "pn-title" title]
+     @div[id: @format["seminar-~a" id] class: @format["col-md-12 pn-seminar ~a" finished]]{
+       @div[id: (seminar-anchor sem) class: "pn-main-informations"]{
+         @a[onclick: "return false;" class: "pn-title" (seminar-title sem)]
          @br[]
-         @span[class: "pn-name" speaker]
-         @span[class: "pn-affiliation" aff]
-         @a[class: "pn-url" target: "_blank" href: link link]
+         @span[class: "pn-name" (seminar-speaker sem)]
```

Comment:
Instead of doing `getter sem` every time, couldn't you pattern-match on `sem` in the very beginning to have bindings for all the field names in the scope?

Reply:
@mhyee one way to do this ^ is to add the line `(match-define (seminar anchor title speaker link aff date room abstract bio) sem)` at the top of the function.

You can also rename the fields, e.g. `sem.anchor`, `sem.title` ...

Either way. I think the current version looks fine.